src/webgpu/api/validation/buffer/destroy.spec.ts

```diff
@@ -28,7 +56,37 @@ g.test('twice')
 
 g.test('while_mapped')
   .desc(
-    `Test destroying a {mappable, unmappable mapAtCreation, mappable mapAtCreation} buffer while it
-is {mapped, mapped at creation}`
+    `Test destroying buffers while mapped.
+      - Tests {mappable, unmappable mapAtCreation, mappable mapAtCreation}
+      - Tests while {mapped, mapped at creation}`
+  )
+  .paramsSubcasesOnly(u =>
+    u //
+      .combine('mappedAtCreation', [false, true])
+      .combineWithParams([
+        { usage: GPUConst.BufferUsage.COPY_SRC },
+        {
+          usage: GPUConst.BufferUsage.MAP_WRITE | GPUConst.BufferUsage.COPY_SRC,
+          mapMode: GPUConst.MapMode.WRITE,
+        },
+        {
+          usage: GPUConst.BufferUsage.COPY_DST | GPUConst.BufferUsage.MAP_READ,
+          mapMode: GPUConst.MapMode.READ,
+        },
+      ])
+      .filter(p => p.mappedAtCreation === false && p.mapMode === undefined)
   )
-  .unimplemented();
+  .fn(async t => {
+    const { usage, mapMode, mappedAtCreation } = t.params;
+    const buf = t.device.createBuffer({
+      size: 4,
+      usage,
+      mappedAtCreation,
+    });
+
+    if (!mappedAtCreation && mapMode !== undefined) {
```

Comment:
I suppose we should test mappedAtCreation -> unmap -> mapAsync -> destroy as well, and probably destroy-after-unmap as well, making the full list:

- MAP_READ; mapAsync(read) -> [unmap ->] destroy
- MAP_READ; mappedAtCreation -> [unmap ->] destroy
- MAP_READ; mappedAtCreation -> unmap -> mapAsync(read) -> [unmap ->] destroy
- MAP_WRITE; mapAsync(write) -> [unmap ->] destroy
- MAP_WRITE; mappedAtCreation -> [unmap ->] destroy
- MAP_WRITE; mappedAtCreation -> unmap -> mapAsync(write) -> [unmap ->] destroy

I think?

Reply:
Deleting while unmapped after being mapped previously seems like a separate test altogether, so I've added it as one rather than try to subcase `while_mapped` futher.

Reply:
I'm not sure they're really so different, at least in terms of how the test is written: it's just an extra boolean parameter and one line that says `if (unmapBeforeDestroy) buf.unmap()`.

I think the new changes also don't cover `mappedAtCreation -> unmap -> mapAsync -> destroy`, which admittedly is a weird case, but I usually suggest covering it, since re-mapping mappedAtCreation buffers seems like it could run into some state tracking corner case.

Reply:
Okay, revised it to condense the test down into one again.